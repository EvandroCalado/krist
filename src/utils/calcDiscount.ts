export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
