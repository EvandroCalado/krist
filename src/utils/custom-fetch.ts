import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
