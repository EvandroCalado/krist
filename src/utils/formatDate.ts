export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
