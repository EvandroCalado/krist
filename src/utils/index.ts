export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
