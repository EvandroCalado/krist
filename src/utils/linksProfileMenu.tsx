import { Heart, MapPin, Package, Settings, UserRound } from 'lucide-react';

export const linksProfileMenu = [
  {
    id: 1,
    label: 'informações pessoais',
    path: 'profile',
    icon: <UserRound size={16} />,
  },
  {
    id: 2,
    label: 'meus pedidos',
    path: 'orders',
    icon: <Package size={16} />,
  },
  {
    id: 3,
    label: 'lista de desejos',
    path: 'wishlist',
    icon: <Heart size={16} />,
  },
  {
    id: 4,
    label: 'meus endereços',
    path: 'addresses',
    icon: <MapPin size={16} />,
  },
  {
    id: 5,
    label: 'configurar',
    path: 'settings',
    icon: <Settings size={16} />,
  },
];
