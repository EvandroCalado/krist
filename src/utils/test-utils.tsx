import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
