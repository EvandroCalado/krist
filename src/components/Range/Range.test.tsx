import { fireEvent, screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { Range } from '.';

describe('Range', () => {
  it('should render default', () => {
    customRender(<Range price={0} setPrice={() => {}} />);

    const input = screen.getByRole('slider');
    const price = screen.getByText('R$0');
    const total = screen.getByText('R$1000');

    expect(input).toBeInTheDocument();
    expect(price).toBeInTheDocument();
    expect(price).toHaveTextContent('R$0');
    expect(total).toBeInTheDocument();
    expect(total).toHaveTextContent('R$1000');
  });

  it('should change price', () => {
    const mockSetPrice = jest.fn();

    jest.mock('react', () => ({
      ...jest.requireActual('react'),
      useState: () => [0, mockSetPrice],
    }));

    customRender(<Range price={0} setPrice={mockSetPrice} />);

    const input = screen.getByRole('slider');

    fireEvent.change(input, { target: { value: 50 } });

    expect(mockSetPrice).toHaveBeenCalledWith(50);
  });

  it('should render a snapshot', () => {
    const { container } = customRender(<Range price={0} setPrice={() => {}} />);

    expect(container).toMatchSnapshot();
  });
});
