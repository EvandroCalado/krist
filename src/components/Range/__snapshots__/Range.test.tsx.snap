// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range should render a snapshot 1`] = `
.c0 {
  width: 100%;
  accent-color: #131118;
  margin-bottom: .8rem;
  cursor: pointer;
}

.c1 {
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: 1.2rem;
}

<div>
  <div
    class=""
  >
    <input
      class="c0"
      max="1000"
      min="0"
      step="1"
      type="range"
      value="0"
    />
    <div
      class="c1"
    >
      <span>
        R$
        0
      </span>
      <span>
        R$1000
      </span>
    </div>
  </div>
</div>
`;
